### DISCUSSION TITLE
Using a covariate only for one of the transition probabilities

### DISCUSSION THREAD

**Discussion by RonEfrat** [TheoMichelot/hmmTMB#26] [Q&A] [answered]
Hi!

I wonder if there is a way (if it even make sense) to use a covariate that affects only the transition probability from s1 to s2, but not from s2 to s1?
In my example, a migrating bird reaches a good habitat and then switches from migratory behavior to stopover behavior. However, the decision to finish the stopover and keep migrating is not based on the habitat. Can I use the habitat at the end of the step as a covariate to explain only the transition from migratory to stopover behavior but not the other way around?

Some more details, if needed: 
Say that a bird is in migratory state. It moves long distances every day in pretty much the same direction, until reaching a habitat which is good for a stopover, so it stays there for a few days during which it changes to short flights to every possible direction. So if it shifted states between steps t1-t2 and t2-t3 (t1-t2 being migration and t2-t3 being stopover), I assume that the habitat at the end of t1 (or the beginning of t2, it is the same) affected that decision. However, after a few days of stopover behavior, it changes again to migratory behavior between steps t7-t8 and t8-t9. So the habitat at the end of t7 is still a "stopover habitat", as this step was still a stopover step, and the habitat at the end of t8 is probably not a "stopover habitat". But, the decision to change behavior was, I assume, not affected by the habitat, but rather by other factors such as how much it has eaten during the stopover days.

Would appreciate your help.
Thanks in advance.

**Comment by TheoMichelot** [comment, marked as answer]
Hi, yes this is possible. You can specify the `formula` argument of `MarkovChain$new()` as a matrix, where each entry is the formula corresponding to a transition probability.

- The formulas in the matrix should be strings, e.g., `"~x1"`, or `"~ x1 * x2"`. The mgcv formula syntax for non-linear and random effects uses quotes, and those should be written as single quotes here, e.g., `"~s(ID, bs = 're')"`.

- The diagonal entries should be `"."` because diagonal transition probabilities are not estimated so they can't have a formula.

Here is an example with a dummy data set, just to illustrate the syntax.
``` r
library(hmmTMB)

data <- data.frame(x1 = rnorm(1000))

# Include effect of x1 on Pr(1 > 2) but not on Pr(2 > 1)
formula_matrix <- matrix(c(".", "~x1",
                           "~1", "."),
                         nrow = 2, byrow = TRUE)

hid <- MarkovChain$new(data = data, 
                       formula = formula_matrix, 
                       n_states = 2)
```

We can then check that the model has a slope parameter for the effect of `x1` on the probability of a transition from state 1 to state 2 (initialised to zero), but not for the other transition probability.
``` r
> hid$coeff_fe()
                       [,1]
S1>S2.(Intercept) -2.197225
S1>S2.x1           0.000000
S2>S1.(Intercept) -2.197225
```

You can find a real example in Section 7.2.2 of [Michelot (2022). _hmmTMB: Hidden Markov models with flexible covariate effects in R_. arXiv preprint, arXiv:2211.14139](https://doi.org/10.48550/arXiv.2211.14139).

Let me know if anything is unclear!

**Reply by RonEfrat** [reply]
Thanks! That is really helpful. 
Now I need to deal with the fact that this change means that the model now fails to converge...
